src/daft-proto/proto/v1/daft.proto

```diff
@@ -0,0 +1,483 @@
+  }
+
+  message FixedShapeImage {
+    ImageMode mode = 1;
+    uint32 height = 2;
+    uint32 width = 3;
+  }
+
+  message Tensor {
+    DataType element_type = 1;
+  }
+
+  message FixedShapeTensor {
+    DataType element_type = 1;
+    repeated uint64 shape = 2;
+  }
+
+  message SparseTensor {
+    DataType element_type = 1;
+    bool indices_offset = 2;
+  }
+
+  message FixedShapeSparseTensor {
+    DataType element_type = 1;
+    repeated uint64 shape = 2;
+    bool indices_offset = 3;
+  }
+}
+
+enum TimeUnit {
+  TIME_UNIT_UNSPECIFIED = 0;
+  TIME_UNIT_NANOSECONDS = 1;
+  TIME_UNIT_MICROSECONDS = 2;
+  TIME_UNIT_SECONDS = 3;
+}
+
+enum ImageMode {
+  IMAGE_MODE_UNSPECIFIED = 0;
+  IMAGE_MODE_L = 1;
+  IMAGE_MODE_LA = 2;
+  IMAGE_MODE_RGB = 3;
+  IMAGE_MODE_RGBA = 4;
+  IMAGE_MODE_L16 = 5;
+  IMAGE_MODE_LA16 = 6;
+  IMAGE_MODE_RGB16 = 7;
+  IMAGE_MODE_RGBA16 = 8;
+  IMAGE_MODE_RGB32F = 9;
+  IMAGE_MODE_RGBA32F = 10;
+}
+
+message Field {
+  string name = 1;
+  DataType data_type = 2;
+}
+
+
+// -------------------------------------------------------------
+//
+//                       EXPRESSIONS
+//
+// -------------------------------------------------------------
+
+message Expr {
+  oneof expr_variant {
+    Column column = 1;
+    Alias alias = 2;
+    AggExpr agg = 3;
+    BinaryOp binary_op = 4;
+    Cast cast = 5;
+    Function function = 6;
+    Over over = 7;
+    WindowFunction window_function = 8;
+    Not not = 9;
+    IsNull is_null = 10;
+    NotNull not_null = 11;
+    FillNull fill_null = 12;
+    IsIn is_in = 13;
+    Between between = 14;
+    List list = 15;
+    Literal literal = 16;
+    IfElse if_else = 17;
+    ScalarFunction scalar_function = 18;
+    Subquery subquery = 19;
+    InSubquery in_subquery = 20;
+    Exists exists = 21;
+  }
+}
+
+message Literal {
+  oneof literal_variant {
+    bool null = 1;
+    bool boolean = 2;
+    string utf8 = 3;
+    bytes binary = 4;
+    FixedSizeBinary fixed_size_binary = 5;
+    int32 int8 = 6;
+    uint32 uint8 = 7;
+    int32 int16 = 8;
+    uint32 uint16 = 9;
+    int32 int32 = 10;
+    uint32 uint32 = 11;
+    int64 int64 = 12;
+    uint64 uint64 = 13;
+    Timestamp timestamp = 14;
+    int32 date = 15;
+    Time time = 16;
+    Duration duration = 17;
+    Interval interval = 18;
+    double float64 = 19;
+    Decimal decimal = 20;
+    Series series = 21;
+    Struct struct = 22;
+  }
+
+  message FixedSizeBinary {
+    bytes value = 1;
+    uint64 size = 2;
+  }
+
+  message Timestamp {
+    int64 value = 1;
+    TimeUnit unit = 2;
+    optional string timezone = 3;
+  }
+
+  message Time {
+    int64 value = 1;
+    TimeUnit unit = 2;
+  }
+
+  message Duration {
+    int64 value = 1;
+    TimeUnit unit = 2;
+  }
+
+  message Interval {
+    int32 months = 1;
+    int32 days = 2;
+    int64 nanoseconds = 3;
+  }
+
+  message Decimal {
+    string value = 1;
+  }
+
+  message Series {
+    string name = 1;
+    DataType dtype = 2;
+    bytes data = 3;
```

Comment:
this is a bit spooky. we likely need a way to serialize series in a reliable way. Likely some kind of other spec like arrow-ipc or something. For now. let's disable this code path and throw an error, and leave a comment here